enterprise/api/filters.py

```diff
@@ -54,3 +54,24 @@ def filter_queryset(self, request, queryset, view):
             queryset = queryset.filter(user_id=request.user.id)
 
         return queryset
+
+
+class EnterpriseCustomerFilterBackend(filters.BaseFilterBackend):
+    """
+    Filter backend for `EnterpriseCustomerViewSet`
+    """
+
+    def filter_queryset(self, request, queryset, view):
+        """
+        Filter out enterprise customer if learner is not linked
+        """
+        enterprise_slug = request.GET.get('slug')
+        if not request.user.is_staff and enterprise_slug:
```

Comment:
This seems reasonable to me, but just want to confirm that we're ok with not applying these filters to staff users

Reply:
The `UserFilterBackend` also only currently applies for non-staff users; I think it's fine to replicate that here.